### PR TITLE
raftstore: move raftstore to components

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Make raftstore as a crate and move it to components.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by BusyJay** [comment]
Cool, what's the compile time difference between master and this branch?

**Comment by overvenus** [comment]
PTAL, thanks!

**Comment by overvenus** [comment]
Compile-time difference:

master:     Finished dev [unoptimized + debuginfo] target(s) in 4m 26s
PR:     Finished dev [unoptimized + debuginfo] target(s) in 3m 47s

**Comment by siddontang** [comment]
maybe we can trace the release compiling time every day

@brson @zhouqiang-cl

File: components/raftstore/src/into_other.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+// Various conversions between types that can't have their own
+// interdependencies. These are used to convert between error_traits::Error and
+// other Error's that error_traits can't depend on.
```
Comment (NingLin-P) [inline components/raftstore/src/into_other.rs]: `error_traits` -> `engine_traits`?

File: tests/Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ required-features = ["failpoints"]
 [[test]]
 name = "integrations"
 path = "integrations/mod.rs"
-required-features = ["tikv/testexport"]
+required-features = ["raftstore/testexport", "tikv/testexport"]
```
Comment (BusyJay) [inline tests/Cargo.toml]: I think this break "make test" as testexport features is not specified in anywhere.

Comment (overvenus) [reply tests/Cargo.toml]: I wasn't able to repro. 

```
 git show --stat | more
commit abe56a3534762081bbfa1010917c8fc8d595f947

$ EXTRA_CARGO_ARGS=--no-run make test &> /dev/null 

$ echo $?
0
```

Comment (BusyJay) [reply tests/Cargo.toml]: What I mean is the integration test cases are not run at all.

**Pull request merged**